src/audio/module_adapter/module/volume/volume.c

```diff
@@ -941,13 +946,25 @@ static int volume_set_config(struct processing_module *mod, uint32_t config_id,
 	cdata = *(const struct ipc4_peak_volume_config *)fragment;
 	cdata.target_volume = convert_volume_ipc4_to_ipc3(dev, cdata.target_volume);
 
+	if ((cdata.channel_id != IPC4_ALL_CHANNELS_MASK) &&
+	    (cdata.channel_id >= SOF_IPC_MAX_CHANNELS)) {
```

Comment:
why is this check not consistent? >= here but just > in the other 2 changes?

Reply:
Here is checked channel_id, in another places channels_count was checked.